web/packages/extension/src/utils.ts

```diff
@@ -184,3 +91,32 @@ export async function getExplicitOptions(): Promise<Options> {
 
     return options;
 }
+
+export const hasAllUrlsPermission = async () => {
+    const allPermissions = await permissions.getAll();
+    if (Object.prototype.hasOwnProperty.call(allPermissions, "origins")) {
```

Comment:
What are you trying to protect against here? `allPermissions.origins?.includes("<all_urls>")` should work.